org.eclipse.draw2d/src/org/eclipse/draw2d/ConnectionEndpointLocator.java

```diff
@@ -73,23 +72,22 @@ private int calculateConnectionLocation(Point loc, Point topLeft, Point center)
 			// Case where m2 is vertical
 			if (loc.y < center.y) {
 				return 3;
-			} else {
-				return 1;
 			}
-		} else if (Math.abs(m2) <= Math.abs(m1)) {
+			return 1;
+		}
+		if (Math.abs(m2) <= Math.abs(m1)) {
 			// Connection start point along left or right side
 			if (loc.x < center.x) {
 				return 4;
 			} else {
 				return 2;
 			}
+		}
+		// Connection start point along top or bottom
+		if (loc.y < center.y) {
+			return 3;
 		} else {
-			// Connection start point along top or bottom
-			if (loc.y < center.y) {
-				return 3;
-			} else {
-				return 1;
-			}
+			return 1;
```

Comment:
Is there a reason to keep the other "else" branches?

Reply:
This was done by the auto clean-up. Now that I see it also the else in line 82 should be removed. Will fix this,.